packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/Events/index.js

```diff
@@ -0,0 +1,288 @@
+        {formatMessage({
+          id: 'Settings.webhooks.form.events',
+          defaultMessage: 'Events',
+        })}
+      </FieldLabel>
+      <StyledTable>{childrenWithProps}</StyledTable>
+    </Flex>
+  );
+};
+
+Root.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+const Headers = ({ getHeaders = getCEHeaders, isDraftAndPublish = false }) => {
+  const { formatMessage } = useIntl();
+  const headers = getHeaders(isDraftAndPublish);
+
+  return (
+    <thead>
+      <tr>
+        <td />
+        {headers.map((header) => {
+          if (header.id === 'app.utils.publish' || header.id === 'app.utils.unpublish') {
+            return (
+              <td
+                key={header.id}
+                title={formatMessage({
+                  id: 'Settings.webhooks.event.publish-tooltip',
+                  defaultMessage: 'This event only exists for content with draft & publish enabled',
+                })}
+              >
+                <Typography variant="sigma" textColor="neutral600">
+                  {formatMessage(header)}
+                </Typography>
+              </td>
+            );
+          }
+
+          return (
+            <td key={header.id}>
+              <Typography variant="sigma" textColor="neutral600">
+                {formatMessage(header)}
+              </Typography>
+            </td>
+          );
+        })}
+      </tr>
+    </thead>
+  );
+};
+
+Headers.propTypes = {
+  getHeaders: PropTypes.func.isRequired,
+  isDraftAndPublish: PropTypes.bool.isRequired,
+};
+
+const Body = ({ providedEvents, isDraftAndPublish }) => {
+  const events = providedEvents || getCEEvents(isDraftAndPublish);
+  const { values, handleChange: onChange } = useFormikContext();
+
+  const inputName = 'events';
+  const inputValue = values.events;
+  const disabledEvents = [];
+
+  const formattedValue = formatValue(inputValue);
+
+  const handleChange = ({ target: { name, value } }) => {
+    let set = new Set(inputValue);
+
+    if (value) {
+      set.add(name);
+    } else {
+      set.delete(name);
+    }
+    onChange({ target: { name: inputName, value: Array.from(set) } });
+  };
+
+  const handleChangeAll = ({ target: { name, value } }) => {
+    let set = new Set(inputValue);
+
+    if (value) {
+      events[name].forEach((event) => {
+        if (!disabledEvents.includes(event)) {
+          set.add(event);
+        }
+      });
+    } else {
+      events[name].forEach((event) => set.delete(event));
+    }
+    onChange({ target: { name: inputName, value: Array.from(set) } });
+  };
+
+  return (
+    <tbody>
+      {Object.keys(events).map((event) => {
+        return (
+          <EventRow
+            disabledEvents={disabledEvents}
+            key={event}
+            name={event}
+            events={events[event]}
+            inputValue={formattedValue[event]}
+            handleChange={handleChange}
+            handleChangeAll={handleChangeAll}
+          />
+        );
```

Comment:
might be better to use `Object.entries`? then you don't have to do `events[event]` for example – wdyt?